src/coreclr/jit/importercalls.cpp

```diff
@@ -2989,7 +2992,25 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                     op1     = op1->AsCall()->gtArgs.GetArgByIndex(0)->GetNode();
                     retNode = op1;
                 }
-                // Call the regular function.
+                else if (op1->OperIs(GT_RET_EXPR))
```

Comment:
Generalize it to if the call wasn't marked as an inline candidate? Or will it have been spilled for that case?

Reply:
I'll have to duplicate recognition logic for that I guess since we only expand a certain shape and the rest should be inlined so the current code is a bit simpler to me

Reply:
Wouldn't it just be:
```cpp
                else if (op1->OperIs(GT_CALL, GT_RET_EXPR))
                {
                    // Skip roundtrip "handle -> RuntimeType -> handle" for
                    // RuntimeTypeHandle.ToIntPtr(typeof(T).TypeHandle)
                    GenTreeCall* call = op1->OperIs(GT_CALL) ? op1->AsCall() : op1->AsRetExpr()->gtInlineCandidate;
```
?

Reply:
Ah, I misunderstood you. Yes, makes sense